pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -339,6 +339,53 @@ func TestEtcdUpdate(t *testing.T) {
 
 }
 
+func TestNoOpUpdates(t *testing.T) {
+	server, registry := NewTestGenericEtcdRegistry(t)
```

Comment:
> although in that case no-op update is not no-op so it will be performed

fair enough. I guess the first one would actually update, and correctly empty the `resourceVersion` in etcd, so a second update should be a true no-op, right?